src/core/action-creator.ts

```diff
@@ -41,38 +42,140 @@ export function createActions(flux: FluxCapacitor) {
+              msg: 'low value must be lower than high'
+            }, {
+              func: ({ range }) => !!range || validators.isString.func(valueOrLow),
+              msg: `value ${validators.isString.msg}`
+            }, {
+              func: (payload, state) => {
+                const navigation = Selectors.navigation(state, field);
+                // tslint:disable-next-line max-line-length
+                return !navigation || navigation.selected
+                  .findIndex((index) => SearchAdapter.refinementsMatch(payload, <any>navigation.refinements[index], navigation.range ? 'Range' : 'Value')) === -1;
+              },
+              msg: 'refinement is already selected'
+            }]
+          }
+        })
+      ],
+
+      switchRefinement: (field: string, valueOrLow: any, high: any = null): Actions.SwitchRefinement => <any>[
+        actions.resetPage(),
+        ...actions.resetRefinements(field),
+        ...actions.addRefinement(field, valueOrLow, high)
+      ],
+
+      resetRefinements: (field?: boolean | string): Actions.ResetPageAndResetRefinements => [
+        actions.resetPage(),
+        action(Actions.RESET_REFINEMENTS, field, {
+          ...metadata,
+          validator: {
+            payload: [{
+              func: () => field === true || typeof field === 'string',
+              msg: 'clear must be a string or true'
+            }, {
+              func: (_, state) => Selectors.selectedRefinements(state).length !== 0,
+              msg: 'no refinements to clear'
+            }, {
+              // tslint:disable-next-line max-line-length
+              func: (_, state) => typeof field === 'boolean' || Selectors.navigation(state, field).selected.length !== 0,
+              msg: `no refinements to clear for field "${field}"`
+            }]
+          }
+        })
+      ],
 
-      search: (query: string = Selectors.query(flux.store.getState())) =>
-        actions.updateSearch({ query, clear: true }),
+      resetPage: (): Actions.ResetPage =>
+        action(Actions.RESET_PAGE, undefined, {
+          ...metadata,
+          validator: {
+            payload: {
+              func: (_, state) => Selectors.page(state) !== 1,
+              msg: 'page must not be on first page'
+            }
+          }
+        }),
 
-      resetRecall: (query: string = null, { field: navigationId, index }: { field: string, index: number } = <any>{}) =>
-        actions.updateSearch({ query, navigationId, index, clear: true }),
+      search: (query: string = Selectors.query(flux.store.getState())): Actions.Search => <any>[
+        actions.resetPage(),
+        ...actions.resetRefinements(true),
+        ...actions.updateQuery(query)
+      ],
+      // tslint:disable-next-line max-line-length
+      resetRecall: (query: string = null, { field, index }: { field: string, index: number } = <any>{}): Actions.ResetRecall => {
+        const resetActions: any = [
+          actions.resetPage(),
```

Comment:
this initial value is equivalent to spreading `actions.search()` so lets use that instead to keep it DRY